packages/blockchain-link/src/workers/solana/index.ts

```diff
@@ -1,20 +1,29 @@
+import { A, F, pipe } from '@mobily/ts-belt';
 import type { Response, AccountInfo } from '@trezor/blockchain-link-types';
 import type * as MessageTypes from '@trezor/blockchain-link-types/lib/messages';
 import { CustomError } from '@trezor/blockchain-link-types/lib/constants/errors';
 import { BaseWorker, ContextType, CONTEXT } from '../baseWorker';
 import { MESSAGES, RESPONSES } from '@trezor/blockchain-link-types/lib/constants';
 import { Connection, PublicKey } from '@solana/web3.js';
+import { TokenInfo } from 'packages/blockchain-link-types/lib';
 
 export type SolanaAPI = Connection;
 
 type Context = ContextType<SolanaAPI>;
 type Request<T> = T & Context;
 
+const TOKEN_PROGRAM_PUBLIC_KEY = new PublicKey('TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA');
+
 const getAccountInfo = async (request: Request<MessageTypes.GetAccountInfo>) => {
     const { payload } = request;
     const api = await request.connect();
 
-    const accountInfo = await api.getAccountInfo(new PublicKey(payload.descriptor));
+    const publicKey = new PublicKey(payload.descriptor);
+
+    const accountInfo = await api.getAccountInfo(publicKey);
+    const tokenInfo = await api.getParsedTokenAccountsByOwner(publicKey, {
```

Comment:
Please check [this commit](https://github.com/vacuumlabs/trezor-suite/pull/17/commits/e0845eaca09ff7222011c15868bb01b6be325f45) from the tx history PR, it gets transactions only if  `payload.details === txs`

We should get `tokens` only if payload.details === `tokens` or `tokenBalances`

Check how its done for other backends please

Reply:
I guess we need to test it, but my logic is that tokens should be `undefined` when the `details` is `basic` and defined when its `tokens` or `tokenBalances`. But I am not sure whether this distinction is actually used or its just some leftover 

HAve you found out the difference between `tokens` and `token-balances` ?